integrated_channels/canvas/exporters/learner_data.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+"""
+Learner data exporter for Enterprise Integrated Channel Canvas.
+"""
+
+from datetime import datetime
+from logging import getLogger
+
+from django.apps import apps
+
+from enterprise.api_client.discovery import get_course_catalog_api_service_client
+from integrated_channels.integrated_channel.exporters.learner_data import LearnerExporter
+
+LOGGER = getLogger(__name__)
+
+
+class CanvasLearnerExporter(LearnerExporter):
+    """
+    Class to provide a Canvas learner data transmission audit prepared for serialization.
+    """
+    def get_learner_data_records(
+            self,
+            enterprise_enrollment,
+            completed_date=None,
+            is_passing=False,
+            **kwargs
+    ):  # pylint: disable=arguments-differ,unused-argument
+        """
+        Return a CanvasLearnerDataTransmissionAudit with the given enrollment and course completion data.
+
+        If completed_date is None, then course completion has not been met.
+
+        If no remote ID can be found, return None.
+        """
+        percent_grade = kwargs.get('grade_percent', None)
+        completed_timestamp = completed_date.strftime("%F") if isinstance(completed_date, datetime) else None
+        if enterprise_enrollment.enterprise_customer_user.user_email is not None:
```

Comment:
I feel like since 99% of the logic of this method is under this if block, why not just make it:
 ```
if enterprise_enrollment.enterprise_customer_user.user_email is None:
    Logger.debug('blah blah blah')
    return None
<Do everything else top level>
```

Reply:
fair points!